tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -3877,3 +3891,449 @@ def test_complex128():
     assert (f_complex64() == test_complex64())
     assert (f_complex128() == test_complex128())
 
+def test_numpy_imag_scalar(language):
+
+    @types('bool')
+    def test_bool(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    @types('int')
+    def test_int(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    @types('int8')
+    def test_int8(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    @types('int16')
+    def test_int16(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    @types('int32')
+    def test_int32(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    @types('int64')
+    def test_int64(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    @types('float')
+    def test_float(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    @types('float32')
+    def test_float32(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    @types('float64')
+    def test_float64(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    @types('complex64')
+    def test_complex64(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    @types('complex128')
+    def test_complex128(a):
+        import numpy as np
+        b = np.imag(a)
+        return b
+
+    import numpy as np
+
+    if sys.platform == 'win32':
+        integer = randint(-2147483648, 2147483647)
+    else:
+        integer = randint(-9223372036854775808, 9223372036854775807)
+    integer8 = np.int8(randint(-128, 127))
+    integer16 = np.int16(randint(-32768, 32767))
+    integer32 = np.int32(randint(-2147483648, 2147483647))
+    integer64 = np.int64(randint(-9223372036854775808, 9223372036854775807))
+    fl = np.float(randint(-2147483648, 2147483647))
+    fl32 = np.float32(randint(-2147483648, 2147483647))
+    fl64 = np.float64(randint(-9223372036854775808, 9223372036854775807))
```

Comment:
can you use your declared variable of number limits instead of literal values?
aka max_float ...